drivers/include/periph/spi.h

```diff
@@ -22,127 +18,116 @@
  * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
  */
 
-#ifndef SPI_H
-#define SPI_H
+#ifndef PERIPH_SPI_H
+#define PERIPH_SPI_H
 
+#include <stdbool.h>
+#include <stddef.h>
 #include <stdint.h>
+#include <limits.h>
 
+#include "periph_cpu.h"
 #include "periph_conf.h"
+#include "periph/gpio.h"
+/* TODO: remove once all platforms are ported to this interface */
+#include "periph/dev_enums.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-/* add guard for the case that no SPI device is defined */
-#if SPI_NUMOF
-
 /**
- * @brief Definition available SPI devices
+ * @brief   Default SPI device access macro
  */
-typedef enum {
-#if SPI_0_EN
-    SPI_0 = 0,          /**< SPI device 0 */
+#ifndef SPI_DEV
+#define SPI_DEV(x)      (x)
 #endif
-#if SPI_1_EN
-    SPI_1,              /**< SPI device 1 */
+
+/**
+ * @brief   Define global value for undefined SPI device
+ */
+#ifndef SPI_UNDEF
+#define SPI_UNDEF       (-1)
 #endif
-#if SPI_2_EN
-    SPI_2,              /**< SPI device 2 */
+
+/**
+ * @brief   Default SPI chip select pin access macro
+ */
+#ifndef SPI_HWCS
+#define SPI_HWCS(x)     (INT_MIN + x)
 #endif
-#if SPI_3_EN
-    SPI_3,              /**< SPI device 3 */
+
+/**
+ * @brief   Define value for unused CS line
+ */
+#ifndef SPI_CS_UNDEF
+#define SPI_CS_UNDEF    (GPIO_UNDEF)
 #endif
-} spi_t;
 
 /**
- * @brief The SPI mode is defined by the four possible combinations of clock polarity and
- *        clock phase.
+ * @brief   Default type for SPI devices
  */
-typedef enum {
-    /**
-     * The first data bit is sampled by the receiver on the first SCK edge. The
-     * first edge of SCK is rising. This is sometimes also referred to as SPI
-     * mode 0, or (CPOL=0, CPHA=0).
-     */
-    SPI_CONF_FIRST_RISING = 0,
-    /**
-     * The first data bit is sampled by the receiver on the second SCK edge. The
-     * first edge of SCK is rising, i.e. the sampling edge is falling. This is
-     * sometimes also referred to as SPI mode 1, or (CPOL=0, CPHA=1).
-     */
-    SPI_CONF_SECOND_RISING = 1,
-    /**
-     * The first data bit is sampled by the receiver on the first SCK edge. The
-     * first edge of SCK is falling. This is sometimes also referred to as SPI
-     * mode 2, or (CPOL=1, CPHA=0).
-     */
-    SPI_CONF_FIRST_FALLING = 2,
-    /**
-     * The first data bit is sampled by the receiver on the second SCK edge. The
-     * first edge of SCK is falling, i.e. the sampling edge is rising. This is
-     * sometimes also referred to as SPI mode 3, or (CPOL=1, CPHA=1).
-     */
-    SPI_CONF_SECOND_FALLING = 3
-} spi_conf_t;
+#ifndef HAVE_SPI_T
+typedef unsigned int spi_t;
```

Comment:
with the default above it might make more sense to make this signed, alternatively, change the SPI_UNDEF and SPI_HWCS macros to something `>0`